icp-deploy.tf

```diff
@@ -62,14 +66,23 @@ module "icpprovision" {
       "docker_password"                 = "${local.registry_creds != "" ? "${replace(local.registry_creds, "/.*:/", "")}" : "" }"
     }
 
+    hooks = {
+      "cluster-preconfig" = [
+        "sudo apt-get update",
+        "sudo apt-get -y install --only-upgrade docker-ce"
```

Comment:
this step seems to assume Ubuntu only ...  we need to support both RHEL and Ubuntu in this terraform.

additionally, @hassenius and I were debating on whether we should install docker as a pre-requisite to the icp deploy module as docker installation may be different in each infrastructure provider.

Reply:
@jkwong888 if it's just those 2, then it shouldn't be hard to add logic that installs one or the other.

Are we planning to support more OSs other than Ubuntu and RHEL? If so, then we probably shouldn't install Docker as it can get hard to maintain.

Reply:
No, as of now, we are only going to support Ubuntu and RHEL.  SLES can be reverse engineered by what we have done so far.

Reply:
my concern with the docker install is actually block device naming conventions in different cloud providers.  a best practice (actually CIS level 1) is that docker volumes are on a separate block device.  depending on which hypervisor a second disk could be named /dev/sdb or /dev/xvdb.  that's why I prefer installing it out at the infrastructure level before calling the ICP install module.  I thought our scripts already handled the second block device though.

Reply:
@jkwong888 say no more. Makes sense

Reply:
The cluster-preconfig step should be taken out from this repo, as it originated in a test we did in the lab.